src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -255,12 +250,7 @@ class GoogleExtensions(
     for {
       maybePet <- directoryDAO.loadPetServiceAccount(PetServiceAccountId(userId, project))
       deletedSomething <- maybePet match {
-        case Some(pet) =>
-          for {
-            _ <- directoryDAO.deletePetServiceAccount(PetServiceAccountId(userId, project))
-            _ <- IO.fromFuture(IO(googleIamDAO.removeServiceAccount(project, toAccountName(pet.serviceAccount.email))))
-          } yield true
-
+        case Some(pet) => IO.fromFuture(IO(removePetServiceAccount(pet))).map(_ => true)
```

Comment:
fyi, you can do `IO.fromFuture(IO(removePetServiceAccount(pet))).as(true)` if you `import cats.implicits._`